### ISSUE TITLE
Search object by attribute with less or greater than

### ISSUE THREAD

**Issue by schneidr** [zorn96/ms_active_directory#82]
Is there any way to search for objects by attributes and using `<=` or `>=` as search operators?

For example, I want to search for all user accounts that have been created within the last seven days. With a regular LDAP query I can do this:

```python
    start_time = datetime.today()  - timedelta(days = 7)
    result = self.search(
        "(&(objectClass=user)(whenCreated>={}))".format(start_time.strftime("%Y%m%d%H%M%S.0Z")),
        ["cn", "displayName", "mail", "o", "whenCreated"]
    )
```

With this library I only see the option to search with the `=` operator. Is there any way to to this with ms_active_directory?

**Comment by ndelo** [comment]
Agreed. It would be nice in general to be able to pass an ldap filter string in a search. I was also looking to filter by the whenCreated attribute in the same way as @schneidr 

If this can already be done, it would be nice to see this reflected in the documentation.

**Comment by zorn96** [comment]
hi! the library currently only supports basic equality comparison.
you can always directly call search functions on the underlying ldap3 connection object for a session stored in its `self.ldap_connection`.

any kind of "take arbitrary filter strng" api will inherently be unable to provide any further utility with strong guarantees, and would likely just need to return raw dicts without deserializing them into a specific AD object (otherwise you could get errors if your filter doesn't return exactly what's expected). but i could look at adding that if that's what's being asked for?
